test/unit/frontend/components/form-row.spec.js

```diff
@@ -0,0 +1,197 @@
+import { mount } from '@vue/test-utils'
+import FormRow from '@/components/FormRow.vue'
+import { expect, vi } from 'vitest'
+
+describe('FormRow', () => {
+    test('mounts with type="checkbox"', async () => {
+        expect(FormRow).toBeTruthy()
+
+        const wrapper = mount(FormRow, {
+            props: {
+                type: 'checkbox'
+            },
+            shallow: true
+        })
+
+        // console.log(wrapper.html())
+
+        expect(wrapper.find('ff-checkbox').exists()).toBe(true)
+        expect(wrapper.find('input[type="file"]').exists()).toBe(false)
+        expect(wrapper.find('ff-dropdown').exists()).toBe(false)
+        expect(wrapper.find('ff-text-input').exists()).toBe(false)
+        expect(wrapper.find('.uneditable').exists()).toBe(false)
+    })
+
+    test('mounts with type="file"', async () => {
+        expect(FormRow).toBeTruthy()
+
+        const wrapper = mount(FormRow, {
+            props: {
+                type: 'file'
+            },
+            shallow: true
+        })
+
+        expect(wrapper.find('ff-checkbox').exists()).toBe(false)
+        expect(wrapper.find('input[type="file"]').exists()).toBe(true)
+        expect(wrapper.find('ff-dropdown').exists()).toBe(false)
+        expect(wrapper.find('ff-text-input').exists()).toBe(false)
+        expect(wrapper.find('.uneditable').exists()).toBe(false)
```

Comment:
I'm slightly concerned that asserting a thing doesn't exist with a non class specific classname is kind of brittle, but not blocking.

Reply:
Resolved in https://github.com/flowforge/flowforge/pull/1201/commits/835855731d4f8661b98627310b4f9884cc4e7289